components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -17682,4 +17983,1197 @@ public void setSubscriptionStatus(String subscriptionStatus) {
+     * @return UUID of the newly created shared policy
+     * @throws APIManagementException
+     */
+    public String addAPISpecificOperationPolicy(String apiUUID, String revisionUUID,
+                                                OperationPolicyData policyData)
+            throws APIManagementException {
+
+        OperationPolicySpecification policySpecification = policyData.getSpecification();
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            try {
+                connection.setAutoCommit(false);
+                String policyID = addAPISpecificOperationPolicy(connection, apiUUID, revisionUUID, policyData, null);
+                connection.commit();
+                return policyID;
+            } catch (SQLException e) {
+                connection.rollback();
+                handleException("Failed to add API specific operation policy " + policySpecification.getName()
+                        + " for API " + apiUUID, e);
+            }
+        } catch (SQLException e) {
+            handleException("Failed to add API specific operation policy " + policySpecification.getName()
+                    + " for API " + apiUUID, e);
+        }
+        return null;
+    }
+
+    private String addAPISpecificOperationPolicy(Connection connection, String apiUUID, String revisionUUID,
+                                                 OperationPolicyData policyData, String clonedPolicyId)
+            throws SQLException {
+
+        String policyUUID = addOperationPolicyContent(connection, policyData);
+
+        String dbQuery;
+        if (revisionUUID != null) {
+            dbQuery = SQLConstants.OperationPolicyConstants.ADD_API_SPECIFIC_OPERATION_POLICY_WITH_REVISION;
+        } else {
+            dbQuery = SQLConstants.OperationPolicyConstants.ADD_API_SPECIFIC_OPERATION_POLICY;
+        }
+
+        PreparedStatement statement = connection.prepareStatement(dbQuery);
+        statement.setString(1, policyUUID);
+        statement.setString(2, apiUUID);
+        statement.setString(3, clonedPolicyId);
+        if (revisionUUID != null) {
+            statement.setString(4, revisionUUID);
+        }
+        statement.executeUpdate();
+        statement.close();
+        return policyUUID;
+    }
+
+    /**
+     * This method is used to populate AM_OPERATION_POLICY table. This will return the policy ID.
+     *
+     * @param connection DB connection
+     * @param policyData Unique Identifier of API
+     * @return UUID of the newly created policy
+     * @throws SQLException
+     */
+    private String addOperationPolicyContent(Connection connection, OperationPolicyData policyData)
+            throws SQLException {
+
+        OperationPolicySpecification policySpecification = policyData.getSpecification();
+        String dbQuery = SQLConstants.OperationPolicyConstants.ADD_OPERATION_POLICY;
+        String policyUUID = UUID.randomUUID().toString();
```

Comment:
Can't we pass policyUUID value to method instead of defining here? Do we really need to return it after add operation?